test/dynamo/test_misc.py

```diff
@@ -3819,6 +3819,67 @@ def fn(x, y):
         res = opt_fn(x, y)
         self.assertTrue(same(ref, res))
 
+    def test_int_neg(self):
+        def int_neg(a, b):
+            x = a.shape[0]
+            y = b.shape[0]
+            return -x * -y * a * b
+
+        torch._dynamo.testing.standard_test(self, int_neg, 2)
+
+    def test_has_attr_torch(self):
+        def fn_custom_get_attr(x, y):
+            if hasattr(torch.ops.aten, "mul"):
+                x = x * y
+                return x
+            else:
+                return x
+
+        with self.assertRaisesRegex(
+            torch._dynamo.exc.Unsupported,
+            "hasattr with custom __getattr__",
+        ):
```

Comment:
This seems like the wrong behavior for `hasattr(torch.ops.aten, "mul")` we should just turn it into a constant.

Reply:
because its a torch op?

Reply:
If the concern is because a guard for an id +/ type check for torch.ops.aten.mul is useless and wasteful, I agree, but I think the solution would be to delegate down the hasattr guard creation to subtypes, not throw out the whole set of logic here? Or are you concerned more with that a generic handler is too much of a pessimism when it comes to guards, and too much of a hassle, and its better to just replicate this logic in a few places?

Reply:
This test has a graph break for "im not sure if torch.ops.aten.mul exists" -- this is a pretty silly graph break.  We should know it exists at compile time.

For my comment above, you are moving some pretty dangerous code from an isolated place (user defined objects) to the base class of all variables.
1) I don't believe that code is correct for all variables.  In some cases wrong.  On other cases too many guards or silly graph breaks (like here).
2) I'm worried about the long term maintainability of moving such a dangerous function to a more visible place.  I think it is likely to lead to bugs in the future when people call that method on things it doesn't support properly.

Reply:
Fair, fair. Let me scope this PR down to just torch checks then, thats where most of the breaks are, to start.